drivers/irqchip/irqchip.c

```diff
@@ -27,8 +28,7 @@ irqchip_device_register(struct device *dev)
 int
 irqchip_irq(void)
 {
-	if (!irqchip_device)
-		panic("Interrupt with no irqchip registered");
+	assert(irqchip);
```

Comment:
Is there a reason you are getting rid of the panic? Assert does not print off any information of failure does it?

Reply:
Look at `debug.h`. It prints the function name, the line number, and the expression that failed.